### ISSUE TITLE
Bigger image previews

### ISSUE THREAD

**Issue by RandomClaim** [searx/searx#2263]
Currently when clicking an image from search resutls the pop-up includes the image at a fixed size.
I propose to either increase that scaling to a somewhat higher fixed value or factor in the original image size so that very small pictures do not become pixelated. 

Currently the preview is too small in my opinion which forces me to visit the website.

**Comment by avinashkanaujiya** [comment]
Yup! Came all the way here to just mention this. Please look into this problem.

**Comment by xnghu** [comment]
i agree. also as a suggested future feature, the ability to filter image results by size or file type would be absolutely wonderful!!

in the meantime, i will slowing be studying the code, as perhaps eventually i will know how to help contribute to the project to make such things a possibility! :P

**Comment by unixfox** [comment]
> i agree. also as a suggested future feature, the ability to filter image results by size or file type would be absolutely wonderful!!

Feel free to open another issue for that :).

> in the meantime, i will slowing be studying the code, as perhaps eventually i will know how to help contribute to t

This issue is most likely only Javascript oriented because it's related to the frontend so if you are good in js then you will probably be able to write some code for that.

**Comment by dalf** [comment]
searx and a browser see two different versions of a same search engine:
* with a browser and javascript enabled, you get all the features. The browser can make an additional HTTP request to display a full screen version of the image.
* without javascript, searx see a simplified version, and can make one or few HTTP requests (to keep the response time low, and to stay stateless). Often the no-javascript version get less care from the different search engines.

Here an example of an google image result without javascript (see https://support.mozilla.org/en-US/questions/994809 ) :
<img width="557" alt="google image" src="https://user-images.githubusercontent.com/1594191/99507237-695ba380-2983-11eb-8986-f905b0fdf1c9.png">

If you click on one image, you go directly to the website. Some images are base64 encoded.

This is what searx sees. 

bing image seems to be broken without javascript inside a browser, but searx manages to get the results.

So most probably and unfortunately, it is difficult to get bigger image previews.

Most probably, because _may be_ for some image search engines, there are some tricks to get the full size image URL with no / few HTTP request(s). As usual for the engines: it requires a careful analysis, and maintenance over time (since a search engine can update its web site any time without notice).

**Comment by vacom13** [comment]
Has this been worked upon?

**Comment by unixfox** [comment]
> Has this been worked upon?

No but feel free to work on it! Just create a new pull request, and we will review your code then add it to the SearX code if it's a correct code.

**Comment by vacom13** [comment]
@unifox I will try 😅

**Comment by Deagan** [comment]
> i agree. also as a suggested future feature, the ability to filter image results by size or file type would be absolutely wonderful!!

Yes, please! 🌊🌊🌊

Here is a related issue/feature request:
https://github.com/searx/searx/issues/1214

**Comment by unixfox** [comment]
Searxng (a popular fork of searx) has bigger image previews on the simple theme (default theme when installing searxng): https://github.com/searxng/searxng
